game/src/core/components/DrawComponent.java

```diff
@@ -66,11 +66,16 @@ public class DrawComponent extends Component {
      * @throws IOException if the given path does not exist
      * @see Animation
      */
-    public DrawComponent(Entity entity, String path) throws IOException {
+    public DrawComponent(final Entity entity, final String path) throws IOException {
         super(entity);
         // fetch available animations
         ClassLoader classLoader = getClass().getClassLoader();
-        File directory = new File(classLoader.getResource(path).getFile());
+        File directory;
+        try {
+            directory = new File(classLoader.getResource(path).getFile());
+        } catch (NullPointerException np) {
```

Comment:
@cagix siehst du eine gute möglichkeit das try-catch mit dem if von unten zu verbinden?

Reply:
Hmmm. Hätte auf try-with-resources getippt, aber `File` ist nicht `Autocloseable`.

Aber wenn ich drüber nachdenke, sollte das hier gehen und sogar lesbar sein:

```java
        try {
            directory = new File(classLoader.getResource(path).getFile());
            animationMap =
                Arrays.stream(directory.listFiles())
                .filter(File::isDirectory)
                        .collect(Collectors.toMap(File::getName, Animation::of));
            currentAnimation(CoreAnimations.IDLE_LEFT);
        } catch (NullPointerException np) {
            throw new FileNotFoundException("Path " + path + " not found.");
        }
```

Die explizite Prüfung auf `!directory.exists() || !directory.isDirectory()` kann eigentlich wegfallen: Wenn das Ding nicht existiert oder kein Verzeichnis ist, wird `directory.listFiles()` entweder `null` liefern oder eine Exception werfen. Und genau die fangen wir unten ... 

Fühlt sich dennoch sehr blöd an, sowas im Ctor zu machen. Hmmm.

Reply:
> Fühlt sich dennoch sehr blöd an, sowas im Ctor zu machen. Hmmm.

da fällt mir auf, das Component wird erst der Entität hinzugefügt und dann kann es immernoch crashen... Eine Entität könnte also ein kaputes DrawComponent haben.
Da müsste vorm thrown noch ein `entity.remove(this)` kommen

Reply:
Oh, da gehst Du jetzt aber sehr tief rein. Wenn wir im Ctor crashen, gibt es das Objekt (die Component) ja gar nicht. Die Frage ist nun tatsächlich, was dann in der Entität gespeichert ist? Kannst Du das mal in einem MWE nachstellen?

Reply:
> Wenn wir im Ctor crashen, gibt es das Objekt (die Component) ja gar nicht.

you are right

>  Die Frage ist nun tatsächlich, was dann in der Entität gespeichert ist? Kannst Du das mal in einem MWE nachstellen?

Der Debugger und ich machen uns an die Arbeit

Reply:
das ist ja seltsames verhalten. d.h. die intialisierung der super-klasse reicht, dass wir eine kaputte referenz kriegen?

@AMatutat wir hatten ja irgendwann schon mal drüber diskutiert. vielleicht ist das auch der zeitpunkt, wo wir dieses automatische/implizite registrieren auflösen sollten und in einen separaten aufruf verlagern sollten (analog zu dem registrieren der systeme in game). das würde zwar etwas mehr code bedeuten pro entität/component, aber den code leichter verständlich machen und auch die testbarkeit erhöhen.

Reply:
Hab daraus #782 gemacht